org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionListAdapter.java

```diff
@@ -160,12 +160,13 @@ public View getView(int position, View convertView, ViewGroup parent) {
                             // references accordingly.
                             if (car.equals(mSelectedCar)) {
                                 mSelectedCar = null;
-                                mSelectedButton.setChecked(false);
-                                mSelectedButton = null;
+                                // Call the callback
+                                mCallback.onDeleteCar(car, mSelectedButton);
+                            }
+                            else{
+                                // Call the callback
+                                mCallback.onDeleteCar(car,mSelectedButton);
```

Comment:
No need to differentiate if the car that have been clicked on is the same as the selected car, anymore. If you click the first time on the selected car, the first case will be triggered and the selected car will be set to `null`. If you click "cancel" in the following dialog and you try to delete it again, always the second case will be triggered, because the selected car is `null` now. So, just call `onDeleteCar` without the if-check.